[MRG]: Better argument names for add_tonic_bias()

kenloi opened the pull request:
Closes #347 
For the add_tonic_bias() method in the network class in network.py, I changed the T arg to tstop. In the method, I replaced all the T instances with the new tstop arg.
The _add_drives_from_params(net) function in drives.py calls the add_tonic_bias() method; thus I changed the args to match my changes to add_tonic_bias().
I didn't find any other instances where the add_tonic_bias() method was used so it should be fixed.

ntolley commented:
Great work @kenloi! I'll review the code soon, but you should check out the error log from travis. It seems the `test_network,py` is failing. I believe this is because you need to replace `T` with `tstop` in the tests as well.

ntolley commented on hnn_core/network.py:
```diff
@@ -913,26 +913,25 @@ def add_tonic_bias(self, *, cell_type=None, amplitude=None,
-            T = tstop
-        if T < 0.:
+        if tstop is None:
+            tstop = self.cell_response.times[-1]
```
Maybe make `self.cell_response.times[-1]` a variable like `t_end`.

ntolley commented:
The error is actually a bit deep in the code base. Since the dictionary key is being changed to `tstop`, it's actually breaking some code downstream:
https://github.com/jonescompneurolab/hnn-core/blob/f84f3cfc6787a5a8eee9aa7404037d6d063c7323/hnn_core/network_builder.py#L391
Unpacks the dictionary such the key is a named argument. This means:
https://github.com/jonescompneurolab/hnn-core/blob/f84f3cfc6787a5a8eee9aa7404037d6d063c7323/hnn_core/cell.py#L433
Is getting `tstop` as an argument, so you'll need to change the variable name there as well.

I'd recommend trying to run some simulations to make sure everything is functioning smoothly. You can copy code from the [gamma example](https://github.com/jonescompneurolab/hnn-core/blob/f84f3cfc6787a5a8eee9aa7404037d6d063c7323/examples/plot_simulate_gamma.py#L90) to test the function in a separate notebook.

jasmainak commented:
@kenloi when you make a fix, you should run tests locally rather than wait for Travis CI to fail. You can do that using:

```py
$ py.test .
```

If you want to be be dropped into the [`pdb`](https://docs.python.org/3/library/pdb.html) debugger when an error happens you can do:

```py
$ py.test . --pdb
```

Doing this has saved me a lot of time and it's why the unit tests exist!

kenloi commented:
@ntolley Thanks for the pointers. I'll go through it again.

@jasmainak Thank you for the suggestion. I'll make sure to run tests locally before submitting.

jasmainak commented:
perfect, let us know when you push the next update!

jasmainak commented:
@kenloi once the PR is good to go for you, can you update the title to have the [MRG] as a prefix? Also, you'll need to update `whats_new.rst`. Since the signature of a public method has been changed, you need to add it to the API section. 

Also, make sure to build the docs locally to see how the `whats_new` page renders:

```sh
$ make html-noplot
```

to build without running examples,

and/or

```sh
$ make html
```

to build everything. The html outputs will be inside the _build/ folder

cjayb commented:
Hey @kenloi your last commit (20da1e5) included a bunch of unwanted files. No worries, here's what I'd do. Make a note of the intended changes for the last commit, then simply cut off the last commit using:

```bash
git reset --hard HEAD~1
```

Then re-implement the intended changes and `git add` them.

I recommend setting up an alias in your `~/.bashrc` or equivalent

```bash
alias gad='git add -u'
```

Then `gad .` stages all _changes_, but doesn't add any previously untracked files. And make it a habit to check `git status` before running `git commit`.

kenloi commented:
@cjayb Thanks for the suggestions. My earlier commit ([48024b9](https://github.com/jonescompneurolab/hnn-core/pull/354/commits/48024b98b6209196d29df9639f7c141928e0487e)), I ran both `git status` and `git diff` to check which files to add and commit. I had't edited any files other than the whats_new.rst file, but then the tests started failing, reporting that it couldn't find the module 'mne'. This was a surprise given that I had passed all the tests in the commit before ([ca91da4](https://github.com/jonescompneurolab/hnn-core/pull/354/commits/ca91da4d18c5bf5959bcd0c05346f7a7d7630ea1)).

I tried looking up the module from my WSL bash shell and got the same ModuleNotFoundError. I opened a python IDE from powershell and tried importing mne there but I also got the same ModuleNotFoundError. Only after running a `pip install mne` and downloading the module into my WSL bash shell environment did my tests start passing locally. I thought maybe I was missing some files so I sent a `git add --all` and pushed another commit and now the tests are failing again.

cjayb commented:
> Only after running a `pip install mne` and downloading the module into my WSL bash shell environment did my tests start passing locally

This makes sense. It's not super well-documented, but the somato-example requires `mne`. So pip-installing it into your dev environment was the right thing to do.

> I thought maybe I was missing some files so I sent a `git add --all`

This was the fatal mistake: the `--all` added the entire `doc/`-folder into the subsequent commit (not just the intended `whats_new.rst`. A lot of cruft got included from your `make html`. You should just do the hard reset I mentioned above (make a copy of whats_new on your desktop or smth, so it's easy to copy the changes back after the reset).

Once you've reset & committed only the desired changes, you'll have to do

```bash
git push -f
```

to force GitHub to accept the branch: the `reset` is going to make a non-linear change the git history. But in this case it's what you want.

kenloi commented:
@jasmainak I've submitted the final commit, updated the api docs, and fixed the title prefix to [MRG]. Please let me know if there's anything else and thank you for the guidance.

@cjayb Thank you for your input!

jasmainak commented on doc/whats_new.rst:
```diff
@@ -135,6 +137,7 @@ People who contributed to this release (in alphabetical order):
 - `Blake Caldwell`_
 - `Christopher Bailey`_
 - `Carmen Kohl`_
+- `Kenneth Loi`_
```
```suggestion
```

this is contributors for 0.1 release. We'll add you to the contributor list for 0.2 :)

jasmainak reviewed (approved):
Looks good other than minor nitpick

kenloi commented:
@jasmainak no problem! I can wait :)

jasmainak commented:
Congratulation @kenloi on your first contribution! 🎉 

what's the next PR? :)

jasmainak commented:
maybe https://github.com/jonescompneurolab/hnn-core/issues/306 is an easy one to knock off and get more comfortable with git while we find you one that is more "intermediate" level

kenloi commented:
Thanks @jasmainak! #306 sounds good, I’ll give it a try.
